js/app.js

```diff
@@ -1 +1,157 @@
-// Your code here
+//MODEL
+var model = {
+  loggedIn: false,
+  userId: '',
+  username: '',
+  categories: [],
+  articles: []
+};
+
+
+//VIEW
+var loginTemplate;
+var categoryTemplate;
+var articleTemplate;
+
+function compileTemplates() {
+  var loginSource = $('#login-template').html();
+  loginTemplate = Handlebars.compile(loginSource);
+
+  var categorySource = $('#categories-template').html();
+  categoryTemplate = Handlebars.compile(categorySource);
+
+  var articleSource = $('#articles-template').html();
+  articleTemplate = Handlebars.compile(articleSource);
+}
+
+function renderLogin() {
+	var display = loginTemplate(model);
+	$('#loginContainer').html(display);
+}
+
+function renderCategories() {
+	$('#categories').html('');
+ 	for (var i = 0; i < model.categories.length; i++) {
+ 		var display = categoryTemplate(model.categories[i]);
+		$('#categories').append(display);
+	}
+}
+
+function renderArticles() {
+	$('#main').html('');
+ 	for (var i = 0; i < model.articles.length; i++) {
+ 		var display = articleTemplate(model.articles[i]);
+		$('#main').append(display);
+	}
```

Comment:
Disclaimer: this is no better or worse, just wanted to show you another way this could have been done:

```js
function renderArticles() {
    var renderedTemplate = '';
    for (var i = 0; i < model.articles.length; i++) {
        renderedTemplate += articleTemplate(model.articles[i]);
    }
    $('#main').html(renderedTemplate);
}
```